src/vsr/replica.zig

```diff
@@ -6492,36 +6532,34 @@ pub fn ReplicaType(
 ///
 /// For each DVC message:
 ///
-/// - The headers all belong to the same hash chain.
+/// - Each header in the DVC is one of:
+///   - "valid": The corresponding header is known. The actual message may be:
+///     - prepared (and WAL is valid), or
+///     - prepared (but WAL is corrupt), or
+///     - never prepared.
+///   - "blank": The replica never prepared the corresponding op, and does not have the header.
+///   - "fault": The replica may have prepared the corresponding op, but it does not know the
+///     header and the entry is corrupt.
+/// - The "valid" headers all belong to the same hash chain.
 ///   - Reason: If multiple replicas with the same canonical log_view disagree about an op, the new
 ///     primary could not determine which is correct.
-///   - Gaps are permitted, but the DVC-sender is responsible for ensuring they do not conceal
-///     chain breaks.
+///   - The DVC-sender is responsible for ensuring blanks/faults do not conceal chain breaks.
 ///   - For example,
-///     - a DVC of 6a,8a is valid (6a/8a belong to the same chain).
-///     - a DVC of 6b,8a is invalid (the gap at 7 conceal a chain break).
+///     - a DVC of 6a,7_,8a is valid (6a/8a belong to the same chain).
+///     - a DVC of 6b,7_,8a is invalid (the gap at 7 conceal a chain break).
 ///     - a DVC of 6b,7b,8a is invalid (7b/8a is a chain break)..
 /// - All pipeline headers present on the replica must be included in the DVC headers.
-///
-/// - If the replica does not possess the oldest known pipeline entry (the "DVC anchor")
-///   (usually 1 + op_head -| pipeline_prepare_queue_max, except it does not need to move backwards
-///   when op_head is truncated), then they should include `op_head -| pipeline_prepare_queue_max`
-///   if it is in the journal.
-///   (By the intersection property, at least one DVC will contain one or the other).
-///   - Reason: The new primary may truncate the entire pipeline (6,7,8,9) due to a gap (6),
-///     but afterwards it still requires a head op to repair/chain backward from.
-///     (According to the intersection property, a gap in the unified pipeline indicates an
-///     uncommitted op).
-///   - For example, given pipeline_prepare_queue_max=3:
-///     - a DVC of 7,8 is invalid if replica.commit_min=5.
-///     - a DVC of 7,8 is valid if replica.commit_min=6.
-///     - a DVC of 5,7,8 is valid. (5,_,7,8)
-///     - a DVC of 5,8 is valid.   (5,_,_,8)
-///     - a DVC of 0,1 is valid.
+///   - When `replica.commit_max ≤ replica.op`,
+///     the DVC must include a valid/blank/fault header for every op in that range.
+///   - When `replica.commit_max > replica.op`, only a single header (`replica.op`) is included.
+///   - The DVC will need a valid `commit_max` to complete, since the entire pipeline may be
+///     truncated, and the new primary still needs a head op.
```

Comment:
This talks about commit_min, right?

Reply:
Ah, no, its me being confused

Reply:
At the end of a new primary's repair, `commit_min == commit_max`.

But immediately after the DVC quorum finishes, it might decide to truncate the entire pipeline.
If we do truncate our entire pipeline, then the new head op == `commit_max`.
In order to repair (or do anything), our head op must always correspond to a valid header in the journal.
To guarantee that the new primary _has_ that valid header, DVCs must always include the valid `commit_max` header (if they are able to provide it — by the intersection property we will at least one).

I rewrote this to (hopefully) clarify:
```
///   - The DVC will need a valid header corresponding to its `commit_max` to complete, since the
///     entire pipeline may be truncated, and the new primary still needs a header for its head op.
```